app/models/identifier.js

```diff
@@ -67,7 +67,11 @@ export default class IdentifierModel extends AbstractValidationModel {
             const changedAttributes = (
               await this.structuredIdentifier
             ).changedAttributes();
-            if (changedAttributes?.localId && !localId.match(/^\d+$/)) {
+            if (
+              localId &&
+              changedAttributes?.localId &&
+              !localId.match(/^\d+$/)
+            ) {
```

Comment:
**Suggestions:** Use the optional chaining (?.) will have the same effect and be more consistent with `changedAttributes?.localId`. 

```suggestion
            if (changedAttributes?.localId && !localId?.match(/^\d+$/)) {
```

Reply:
In this case it will not have the same effect. 
- For an empty string `?.` will succeed, after which the `match` fails due to the `+` in the regex. The negation will cause the result to be truthy. This turns the whole condition truthy, which triggers a validation error.
- For `null` the behaviour also changes, although I am not 100% sure on why. I _think_ the `changedAttributes?.localId` succeeds when `localId` is `null` (but not for `undefined`?) and the match statement succeeds in both cases as `?.` results in `undefined` which is in turn negated to truthy.

In trying to to understand the above behaviour a more logical solution popped in my head. Namely, changing the `+` in the match regex to a `*` so that 0 or more numbers are valid instead of 1 or more.
```suggestion
            if (changedAttributes?.localId && !localId.match(/^\d*$/)) {
```
My test for `null` as value for `localId` still fails with this change, but that does not seem to a scenario I can cause via the frontend.

Reply:
@mirdono I like the `/^\d*$/` solution! :+1: